network/src/peer_store/ban_list.rs

```diff
@@ -66,10 +67,9 @@ impl BanList {
     /// Whether the address is banned
     pub fn is_addr_banned(&self, addr: &Multiaddr) -> bool {
         let now_ms = unix_time_as_millis();
-        if let Ok(ip_port) = addr.extract_ip_addr() {
-            return self.is_ip_banned_until(ip_port.ip, now_ms);
-        }
-        false
+        multiaddr_to_socketaddr(addr)
```

Comment:
If we can't check whether a address is banned we should ban the address or raise an error.

Reply:
I follow the original logic here: if we can not extract ip addr from the multiaddr, retur false:

```
  if let OK(ip_port) = addr.extract_ip_addr() {
 }
 false
```